src/Asset/TagRenderer.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Asset;
+
+use Symfony\Component\Asset\Packages;
+
+final class TagRenderer
+{
+    private $entrypointLookup;
+
+    private $manifestLookup;
+
+    private $packages;
+
+    public function __construct(EntrypointLookup $entrypointLookup, ManifestLookup $manifestLookup, Packages $packages = null)
+    {
+        $this->entrypointLookup = $entrypointLookup;
+        $this->manifestLookup = $manifestLookup;
+        $this->packages = $packages;
+    }
+
+    public function renderWebpackScriptTags(string $entryName, string $packageName = null): string
+    {
+        $scriptTags = [];
+        foreach ($this->entrypointLookup->getJavaScriptFiles($entryName) as $filename) {
+            $scriptTags[] = sprintf(
+                '<script src="%s"></script>',
+                $this->getAssetPath($filename, $packageName)
+            );
+        }
+
+        return implode('', $scriptTags);
+    }
+
+    public function renderWebpackLinkTags(string $entryName, string $packageName = null): string
+    {
+        $scriptTags = [];
+        foreach ($this->entrypointLookup->getCssFiles($entryName) as $filename) {
+            $scriptTags[] = sprintf(
+                '<link rel="stylesheet" href="%s" />',
+                $this->getAssetPath($filename, $packageName)
```

Comment:
this is missing HTML escaping.

Reply:
You're absolutely right. But, this is tricky: the asset path needs the equivalent of `html_attr` escaping from Twig. However, this bundle currently does not rely on Twig, and the logic for this escaping is too big to copy (https://github.com/twigphp/Twig/blob/84b2eb541026dbd39590ba80269ae7676b1bfc46/lib/Twig/Extension/Core.php#L1033) and it's not a good idea anyways, since it's security-related.

The only solution I can think of is to couple the bundle to Twig (no huge deal), and either use that `twig_escape_filter` function directly (bit hacky) or render Twig templates inside of `TagRenderer`.

Any other ideas?

Reply:
well, at first, you can still apply normal PHP escaping. As you put this inside a quoted HTML attribute, things are fine (most of the escaping performed by `html_attr` is to allow to use it outside quotes, so for attribute names or unquoted attribute values).

Not doing any escaping can lead to broken HTML here

Reply:
Ah, ok. I've just used `htmlentities` and added a test. If you know of any issues with that, please let me know!